guardian/typerighter
# Add rule check UI

NB: depends on #445 – please review that first!

## What does this change?

A first pass at a rule checking UI.

The intent is to provide a rule creator with quick feedback by checking lots of content with their rule. By showing these matches, it should be possible to surface false positives without resorting to other tooling, or publishing the rule and checking content in our CMS.

![test-rule](https://github.com/guardian/typerighter/assets/7767575/c2886fa2-3b9c-4248-9be5-29270a4933f1)

A few notes:
- I've added it on a separate page, accessible via the 'wrench' icon in the rules table, per #445. My feeling is that this page should probably just be the edit mode, and we should remove the sidebar for rule editing, but I haven't done that in this PR as it didn't feel necessary to get this work in. WDYT?
- Aware that the matches could do with more behaviour – correct/review/amend semantics, see suggestions, etc., as well as links to the relevant article, etc. More to come.

## How to test

Have a play with a few rules, changing the CAPI query and the rule pattern. Does the testing behave as expected?

You'll need to run both checker and manager services for this to work – `./script/start` should do the job.
